test/development/typescript-plugin/metadata/warn-no-type.test.ts

```diff
@@ -0,0 +1,78 @@
+import type { PluginLanguageService } from '../test-utils'
+import {
+  getPluginLanguageService,
+  getTsFiles,
+  NEXT_TS_ERRORS,
+} from '../test-utils'
+import { join } from 'path'
+
+const fixturesDir = join(__dirname, 'app/warn-no-type')
+
+describe('typescript-plugin - metadata - warn-no-type', () => {
+  let languageService: PluginLanguageService
+
+  beforeAll(() => {
+    languageService = getPluginLanguageService(__dirname)
+  })
+
+  it('should not have diagnostics for metadata with type', () => {
+    const hasTypeDir = join(fixturesDir, 'metadata', 'has-type')
+    const tsFiles = getTsFiles(hasTypeDir)
+
+    for (const tsFile of tsFiles) {
+      const diagnostics = languageService.getSemanticDiagnostics(tsFile)
+      expect(diagnostics.length).toBe(0)
```

Comment:
```suggestion
      expect(diagnostics).toEqual([])
```

You'd want to know what diagnostics you get if it unexpectedly reports diagnostics.